services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java

```diff
@@ -84,20 +86,30 @@ static KeyStore createMergedTruststore(String[] truststores, boolean trustStoreI
             includeDefaultTruststore(truststore);
         }
 
-        mergeFiles(truststores, truststore, true);
+        List<String> discoveredFiles = new ArrayList<>();
+        mergeFiles(truststores, truststore, true, discoveredFiles);
+        if (!discoveredFiles.isEmpty()) {
+            LOGGER.infof("Found the following truststore files under directories specified in the truststore paths %s",
+                    discoveredFiles);
+        }
         return truststore;
     }
 
-    private static void mergeFiles(String[] truststores, KeyStore truststore, boolean topLevel) {
+    private static void mergeFiles(String[] truststores, KeyStore truststore, boolean topLevel, List<String> discoveredFiles) {
         for (String file : truststores) {
             File f = new File(file);
             if (f.isDirectory()) {
-                mergeFiles(Stream.of(f.listFiles()).map(File::getAbsolutePath).toArray(String[]::new), truststore, false);
+                mergeFiles(Stream.of(f.listFiles()).map(File::getAbsolutePath).toArray(String[]::new), truststore, false, discoveredFiles);
             } else {
                 if (file.endsWith(".p12") || file.endsWith(".pfx")) {
                     mergeTrustStore(truststore, file, loadStore(file, PKCS12, null));
+                    if (!topLevel) {
+                        discoveredFiles.add(f.getAbsolutePath());
+                    }
                 } else {
-                    mergePemFile(truststore, file, topLevel);
+                    if (mergePemFile(truststore, file, topLevel) && !topLevel) {
+                        discoveredFiles.add(file);
```

Comment:
I'm sorry, missed this. Last comment, I promise. ;)
```suggestion
                        discoveredFiles.add(f.getAbsolutePath());
```
to align with pkcs12 above.

Reply:
Sure - that's the elusive consistency I was shooting for :)